taskcluster/android_taskgraph/target_tasks.py

```diff
@@ -30,7 +30,7 @@ def index_exists(index_path, reason=""):
 def target_tasks_nightly(full_task_graph, parameters, graph_config):
     def filter(task, parameters):
         build_type = task.attributes.get("build-type", "")
-        return build_type in ("nightly", "focus-nightly", "fenix-nightly")
+        return build_type in ("nightly", "focus-nightly", "fenix-nightly", "nightly-firebase")
```

Comment:
Hey! I had a closer look at the `build-type`s we define. I think it would make more sense to use `focus-nightly-firebase` here: 

```suggestion
        return build_type in ("nightly", "focus-nightly", "fenix-nightly", "focus-nightly-firebase")
```

Builds like this one get a `build-type` automatically assigned: 

https://github.com/mozilla-mobile/firefox-android/blob/d89cf09f1f4866bbd14ecab4d2e29f98f5006d5a/taskcluster/ci/build-apk/kind.yml#L253-L256

https://github.com/mozilla-mobile/firefox-android/blob/d89cf09f1f4866bbd14ecab4d2e29f98f5006d5a/taskcluster/android_taskgraph/transforms/build_apk.py#L18-L24

To me, we should update the `ui-test-apk` kind to use `focus-nightly-firebase` instead of defining another build type:

https://github.com/mozilla-mobile/firefox-android/blob/d89cf09f1f4866bbd14ecab4d2e29f98f5006d5a/taskcluster/ci/ui-test-apk/kind.yml#L129-L133

This way, we don't use `nightly-firebase` which is ambiguous. Does `nightly-firebase` refer to focus, fenix, both, or something else? 

Let me know what you think!

Reply:
I'm confused on the context of `focus` here. The `startup-test` task was originally intended for `Fenix` and migrated over. I changed the `build-type` as the build being pushed to Firebase Test Lab was failing (https://bugzil.la/1819153) as it seemingly was optimized. I changed the `build-type` to `nightly-firebase` (as it disabled optimizations defined here https://github.com/mozilla-mobile/firefox-android/blob/20d392be417d46cb0c1e7afc0394d289720495bd/taskcluster/ci/build-apk/kind.yml#L271). I'm not sure I understand what is proposed here @JohanLorenzo :)

Reply:
Oh I missed this test: 

https://github.com/mozilla-mobile/firefox-android/blob/d89cf09f1f4866bbd14ecab4d2e29f98f5006d5a/taskcluster/ci/ui-test-apk/kind.yml#L271-L275

In which case, the filter should look like this: 

```suggestion
        return build_type in ("nightly", "focus-nightly", "fenix-nightly", "focus-nightly-firebase", "fenix-nightly-firebase")
```

If we don't need `focus` UI tests, then I'd recommend going with 


```suggestion
        return build_type in ("nightly", "focus-nightly", "fenix-nightly", "fenix-nightly-firebase")
```

I'm happy to chat over a Zoom call, if that helps 🙂